R/codemeta_readme.R

```diff
@@ -1,153 +1,164 @@
-  } else {
-    NULL
-  }
-
+  get_badge_links_matching(readme, "travis|appveyor|circleci|codecov|coveralls")
 }
 
+# guess_devStatus --------------------------------------------------------------
+
 ## Either repostatus.org or lifecycle badge
 guess_devStatus <- function(readme) {
+
+  get_badge_links_matching(readme, "repostatus\\.org|lifecycle")
+}
+
+# get_badge_links_matching -----------------------------------------------------
+get_badge_links_matching <- function(readme, pattern) {
+
   badges <- extract_badges(readme)
-  status_badge <- badges[grepl("repostatus.org", badges$link)|
-                           grepl("lifecycle", badges$link),]
-  if (!is.null(status_badge)) {
-    if(nrow(status_badge) >0){
-      status_badge$link
-    }
-  } else {
-    NULL
+
+  if (length(links <- grep(pattern, badges$link, value = TRUE))) {
+
+    links
   }
+  # else NULL implicitly
+}
 
+# get_pkg_name -----------------------------------------------------------------
+get_pkg_name <- function(entry) {
 
+  if (is.null(result <- entry$pkgname)) "" else result
 }
 
+# .ropensci_reviews ------------------------------------------------------------
 # looks for a rOpenSci peer review badge
+.ropensci_reviews <- function() {
 
-get_pkg_name <- function(entry){
-  if(is.null(entry$pkgname)){
-    ""
-  }else{
-    entry$pkgname
-  }
-}
+  url_onboarded_json <- "https://badges.ropensci.org/json/onboarded.json"
+
+  reviews <- jsonlite::read_json(url_onboarded_json)
 
-.ropensci_reviews <- function(){
-  reviews <-  jsonlite::read_json("https://badges.ropensci.org/json/onboarded.json")
-  reviews <- tibble::tibble(review = purrr::map_dbl(reviews, "iss_no"),
-                            package = purrr::map_chr(reviews, get_pkg_name))
+  reviews <- tibble::tibble(
+    review = purrr::map_dbl(reviews, "iss_no"),
+    package = purrr::map_chr(reviews, get_pkg_name)
+  )
 }
 
 ropensci_reviews <- memoise::memoise(.ropensci_reviews)
 
+# guess_ropensci_review --------------------------------------------------------
 guess_ropensci_review <- function(readme) {
-  badges <- extract_badges(readme)
-  badge <- badges[grepl("github.com/ropensci/onboarding/issues/", badges$link),]$link
-  if (length(badge) >= 1) {
-    review <-
-      gsub(".*https://github.com/ropensci/onboarding/issues/", "", badge)
-    review <- as.numeric(review)
-    reviews <- ropensci_reviews()
-    if(review %in% reviews$review){
-      list("@type" = "Review",
-           "url" = paste0("https://github.com/ropensci/onboarding/issues/",
-                          review),
-           "provider" = "http://ropensci.org")
-    } else{
-      NULL
-    }
-  } else {
-    NULL
+
+  url <- "github.com/ropensci/onboarding/issues/"
+
+  badges <- get_badge_links_matching(readme, url)
+
+  if (is.null(badges)) {
+
+    return(NULL)
   }
 
+  review <- as.numeric(stringr::str_remove(badges, paste0(".*https://", url)))
+
+  # TODO: What to do with more than one review or is that not possible?
```

Comment:
more than one review from rOpenSci? not possible at the moment

Reply:
Ok, then I will remove the comment.

Reply:
But good for me to keep in mind if we re-review packages in the future 😺